### DISCUSSION TITLE
installation issues / launching web app

### DISCUSSION THREAD

**Discussion by rickhooy** [nextpyp/.github#2] [Q&A] [answered]
Relatively naive question here. How do I actually launch the app online? 
https://localhost
https://[hostname]
https://[IPv4]

Results in 'unable to connect' on standard web page. 

What am I not configuring correctly?

-Rick

Details: 
I followed the installation directions as best as I could for Ubuntu20 running nextPYP Standalone version (0.5.0)

installed at /usr/local/nextPYP with rwx permissions for nextpyp_user (non-root)

By all accounts the instance is started: 
$ sudo PYP_CONFIG=/usr/local/nextPYP/config.toml -u nextpyp_user /usr/bin/nextpyp-start
OUTPUT: 
Using configuration at /usr/local/nextPYP/config.toml
Reading config.toml using CLI tool ...
Configuring environment ...
Starting singularity container ...
INFO:    instance started successfully

During install, I have tried using the default domain name and 'localhost': 
sudo PYP_USER=nextpyp_user PYP_DOMAIN=localhost ./install
sudo PYP_USER=nextpyp_user ./install
% no issues during install in either case

The config.toml file: (ignore the ' preceding commented (#) sections; added here for visualization purposes only)

[pyp]
'# path to the PYP container file, from a SLURM compute node's point of view
container = '/usr/local/nextPYP/pyp.sif'

'# fast storage for temporary files, ideally local to the compute node
scratch = '/scratch/nextpyp_cache'

'# local directories
binds = ['/data/nextpyp_data']

'#[slurm] I've tried using the config file with and without this section commented

'# user account for the SLURM login node
'#user = '[username]'

'# hostname for the SLURM login node
'#host = '[hostname]'

'# maximum memory (in GiB) available on SLURM compute nodes
'#memoryPerNode = 2

'# maximum CPU cores available on SLURM compute nodes
'#cpusPerNode = 2

[web]
'# storage space for website files, database, etc
'# for best performance, this should be on a filesystem local to the web server
localDir = '/usr/local/nextPYP/local'
'#localDir = '/scratch/nextpyp_cache/local'

'# storage space for files shared between the web server and the SLURM cluster
'# this area should have a lot of available space for large files
sharedDir = '/usr/local/nextPYP/shared'

'# URL of the web server, from a SLURM compute node's point of view
webhost = 'https://[hostname]'
% hostname == output from command line: $ hostname

'# authorization
auth = 'login'

'# directories for workflow files
workflowDirs = ['/usr/local/nextPYP/workflows']

**Comment by abartesaghi** [comment]
Try connecting to `http://localhost:8080` from your browser.

**Reply by rickhooy** [reply]
Tried. No luck. Also attempted to look up which port was opened by nextpyp. Nothing obvious.

**Comment by asarnow** [comment]
@rickhooy It does run on 8080. ~~Do you have a firewall? The installer doesn't open any firewall ports.~~

Sorry Alberto, I had a typo when searching for `firewall-cmd`.

**Reply by abartesaghi** [reply]
If `firewalld` is running, the installer should open http & https ports and output `Configured firewall to allow HTTP/HTTPs traffic` during installation.

**Reply by rickhooy** [reply]
Thanks for the responses! I see - I'll check if firewall is running and look for mentions of firewall during reinstallation

**Reply by abartesaghi** [reply]
This [response](https://github.com/orgs/nextpyp/discussions/3#discussioncomment-7412943) may be relevant here as well.

**Reply by rickhooy** [reply]
Reinstalled: 
"No firewall running, skipped configuration"
Everything else looks fine. 
Changed config.toml file to point to http://localhost:8080 under the [web] section.
Added a [standalone] section to the config file.
Still no web app. 

Another naive question: does the startrprox need to be started manually? Based on the install instructions I start the instance with:
sudo PYP_CONFIG=/usr/local/nextPYP/config.toml -u nextpyp_user /usr/bin/nextpyp-start
Does the nextPYP-rprox start with this command also?

I see in "Step 7" for auto starting nextPYP on boot it looks like nextPYP and nextPYP-rprox are started when this is configured. I did not opt to follow Step 7 when I installed 

In attempting to start nextPYP-rprox directly
sudo PYP_CONFIG=/usr/local/nextPYP/config.toml -u nextpyp_user /usr/bin/nextpyp-startrprox
RETURN: 
Usage:
   startrprox <example.org | localhost | 12.34.65.78>

If applicable, how would I start the nextPYP_rprox? Just looking for an example

**Reply by abartesaghi** [reply]
~~There is no need to manually start the reverse proxy, this is handled internally by the `nextpyp-start` command.~~
Do you see any errors in the file `/usr/local/nextPYP/local/logs/micromon`?

**Reply by rickhooy** [reply, marked as answer]
Web app is now finally opened. The logs did lead to the eventual problem. 

/usr/local/nextPYP/local/logs/micromon <-- this file didn't actually exist

/usr/local/nextPYP/local/logs$ ls
total 8.0K
-rw-rw-r-- 1 nextpyp_user nextpyp_user  22 Oct 28 15:16 hostprocessor
-rw-rw-r-- 1 nextpyp_user nextpyp_user 147 Oct 28 15:16 init.log

$ cat hostprocessor
HostProcessor started

$ cat init.log
nextPYP initializing container ...
local directory is /usr/local/nextPYP/local
/opt/micromon/init.sh: line 20: **/tmp/mongod.conf:** Permission denied

Changed ownership of /tmp/mongod.conf to nextpyp_user

Now web app opens on localhost:8080

So my issue mostly stemmed from nextpyp read/write access to root directory folders for the non-root nextpyp_user.
